code/game/machinery/camera/presets.dm

```diff
@@ -43,9 +43,9 @@
 	var/number = 0 //camera number in area
 
 //This camera type automatically sets it's name to whatever the area that it's in is called.
-/obj/machinery/camera/autoname/New()
+/obj/machinery/camera/autoname/Initialize(mapload)
 	..()
-	spawn(10)
+	if(mapload)

```

Comment:
I REALLY need to rethink this though, because you'll have to make all the child Initialize calls aware of this and that's not kosher